contrib/envconfig/client_config_load.go

```diff
@@ -0,0 +1,412 @@
+		return ClientConfig{}, fmt.Errorf("failed parsing config: %w", err)
+	}
+	return conf, nil
+}
+
+// LoadClientConfigProfileOptions are options for [LoadClientConfigProfile].
+type LoadClientConfigProfileOptions struct {
+	// Override the file path to use to load the TOML file for config. Defaults to TEMPORAL_CONFIG_FILE environment
+	// variable or if that is unset/empty, defaults to [os.UserConfigDir]/temporal/temporal.toml. If ConfigFileData is
+	// set, this cannot be set and no file loading from disk occurs. Ignored if DisableFile is true.
+	ConfigFilePath string
+
+	// TOML data to load for config. If set, this overrides any file loading. Cannot be set if ConfigFilePath is set.
+	// Ignored if DisableFile is true.
+	ConfigFileData []byte
+
+	// Specific profile to use after file is loaded. Defaults to TEMPORAL_PROFILE environment variable or if that is
+	// unset/empty, defaults to "default". If either this or the environment variable are set, load will fail if the
+	// profile isn't present in the config. Ignored if DisableFile is true.
+	ConfigFileProfile string
+
+	// If true, will error if there are unrecognized keys.
+	ConfigFileStrict bool
+
+	// If true, will not do any TOML loading from file or data. This and DisableEnv cannot both be true.
+	DisableFile bool
+
+	// If true, will not apply environment variables on top of file config for the client options, but
+	// TEMPORAL_CONFIG_FILE and TEMPORAL_PROFILE environment variables may still by used to populate defaults in this
+	// options structure.
+	DisableEnv bool
+
+	// Override the environment variable lookup. If nil, defaults to [EnvLookupOS].
+	EnvLookup EnvLookup
+}
+
+// LoadClientConfigProfile loads a specific client config profile from file and then applies environment variable
+// overrides. This will not fail if the config file does not exist. This is effectively a shortcut for
+// [LoadClientConfig] + [ClientConfigProfile.ApplyEnvVars]. See [LoadClientOptionsOptions] and [ClientConfigProfile] on
+// how files and environment variables are applied.
+//
+// WARNING: Environment configuration is currently experimental.
+func LoadClientConfigProfile(options LoadClientConfigProfileOptions) (ClientConfigProfile, error) {
+	if options.DisableFile && options.DisableEnv {
+		return ClientConfigProfile{}, fmt.Errorf("cannot disable file and env")
+	}
+
+	var prof ClientConfigProfile
+
+	// If file is enabled, load it and find just the profile
+	if !options.DisableFile {
+		// Load
+		conf, err := LoadClientConfig(LoadClientConfigOptions{
+			ConfigFilePath:   options.ConfigFilePath,
+			ConfigFileData:   options.ConfigFileData,
+			ConfigFileStrict: options.ConfigFileStrict,
+			EnvLookup:        options.EnvLookup,
+		})
+		if err != nil {
+			return ClientConfigProfile{}, err
+		}
+		// Find profile or fail. The one situation where we don't fail here is if the profile is not user-set
```

Comment:
This comment took me a sec to understand, I think something like this could be easier to understand
```suggestion
		// Find user-set profile or use default. Fail if user uses invalid profile.
```

Reply:
Updated comment